internal/indexmanagement/scripts.go

```diff
@@ -50,6 +51,31 @@ def deleteIndices(index):
     sys.stdout = original_stdout
     return False
 
+#here namespace is either a string or array of strings, minAge is also a string in the unit of days, e.g., '3d'
+def deleteByQuery(index, namespace, minAge):
+  defaultAge='7d'
+  original_stdout = sys.stdout
+  try:
+    es_client = getEsClient()
+    s = Search(using=es_client)
+    #Construct the dsl-query of delete_by_query() body
+    if namespace != None and minAge == None:
+      s = s.query('terms', kubernetes__namespace_name=namespace) #Or just #Q('terms',  tags=['kubernetes.namespace_name', namespace])
+    elif namespace == None and minAge != None:
+      s = s.filter('range', **{'@timestamp': {'lt': 'now-{}'.format(minAge)}})
+    elif namespace == None and minAge == None:
+      s = s.filter('range', **{'@timestamp': {'lt': 'now-{}'.format(defaultAge)}})
+    elif namespace != None and minAge != None:
+      s = s.query('terms', kubernetes__namespace_name=namespace)
+      s = s.filter('range', **{'@timestamp': {'lt': 'now-{}'.format(minAge)}})
```

Comment:
This much filtering is not required. This is complicating the code.
I would suggest you to define a query as:
`body = {
  "query" : {
    "terms": {
      "kubernetes.namespace_name": []        // Namespace
    }`

The above query would be used when `minAge` is not defined. Which suggests to delete all data belonging to that namespace.

Similarly, if `minAge` is defined then you can append the filter part to the above query, in which case, the body will look like
`body = {
  "query": {
    "bool": {
      "must": [
        {
          "terms": {
            "kubernetes.namespace_name": []      // Namespace
          }
        }
      ],
      "filter": [
        {
          "range": {
            "@timestamp": { "lt": "now-$age" }    // $age=MinAge
          }
        }
      ]
    }
  }
}`

Reply:
Does this means that there should always be a namespace_name given? because otherwise it would delete the entire index if no namespace nor minAge is given. (only those entries older than defaultAge, of course)
In this case I shouldn't check for this condition?
`if namespace == None and minAge == None:
      s = s.filter('range', **{'@timestamp': {'lt': 'now-{}'.format(defaultAge)}})`

Reply:
If namespace_name is not given then `delete_by_query` function itself will be never called right. IMO there is no use of checking and generating dynamic body. WDYT?

Reply:
I agree.. I asked because you wrote it as a condition in your spike document. It makes more sense this way!

Reply:
That was mentioned in Spike document just to list all possible cases. You can implement in a way which is making more sense.